cpp/tests/experimental/spatial/linestring_intersection_test.cu

```diff
@@ -30,12 +34,126 @@
 #include <rmm/exec_policy.hpp>
 #include <rmm/mr/device/pool_memory_resource.hpp>
 
+#include <thrust/binary_search.h>
+#include <thrust/sort.h>
+
 #include <initializer_list>
 #include <type_traits>
 
 using namespace cuspatial;
 using namespace cuspatial::test;
 
+/**
+ * @brief Perform sorting to the intersection result
+ *
+ * The result of intersection result is non-determinisitc. This algorithm sorts
+ * the geometries of the same types and the same list and makes the result deterministic.
+ *
+ * The example below contains 2 rows and 4 geometries. The order of the first
+ * and second point is non-deterministic.
+ * [
+ *  [Point(1.0, 1.5), Point(0.0, -0.3), Segment((0.0, 0.0), (1.0, 1.0))]
+ *   ^                ^
+ *  [Point(-3, -5)]
+ * ]
+ *
+ * After sorting, the result is deterministic:
+ * [
+ *  [Point(0.0, -0.3), Point(1.0, 1.5), Segment((0.0, 0.0), (1.0, 1.0))]
+ *   ^                 ^
+ *  [Point(-3, -5)]
+ * ]
+ *
+ * This function invalidates the input @p result and return a copy of sorted results.
+ */
+template <typename T, typename IndexType, typename type_t>
+linestring_intersection_result<T, IndexType> segment_sort_intersection_result(
```

Comment:
I can see us not wanting this to be only a test utility down the road.

Reply:
#858